app/goal_routes.py

```diff
@@ -0,0 +1,128 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.task_routes import validate_task
+
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+def validate_goal(id):
+    try:
+        id = int(id)
+    except:
+        abort(make_response({"message": f"Goal {id} is invalid"}, 400))
+
+    goal = Goal.query.get(id)
+
+    if not goal:
+        abort(make_response({"message": f"Goal {id} not found"}, 404))
+
+    return goal
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    if request.method == "POST":
```

Comment:
No need to check if the request method is POST, since you specified on line 242with the decorator that this route create_goal() is for POST requests.